### ISSUE TITLE
Error: buildx failed with: ERROR: failed to receive status: rpc error: code = Unavailable desc = closing transport due to: connection error: desc = "error reading from server: EOF", received prior goaway: code: NO_ERROR, debug data: "graceful_stop"

### ISSUE THREAD

**Issue by PanDeOps** [docker/buildx#2789]
**Contributing guidelines**

- [ ] I've read the [contributing guidelines](https://github.com/docker/buildx/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree

**I've found a bug and checked that ...**

- [ ] ... the documentation does not mention anything about my problem

- [ ] ... there are no open or closed issues that are related to my problem

**Description**

Greetings

On own ubuntu server via github CI when multibuilding containers (4 pcs) at the end of the docker/build-push-action@v6 module one of the processes crashes with an error on completion, this happens quite often when first running with git action..:
```bash
...
#12 pushing layers
#12 ...

#13 [auth] ***/backend:pull,push token for registry-1.docker.io
#13 DONE 0.0s

#12 exporting to image
#12 ...

#14 [auth] ***/backend:pull,push ***/app:pull token for registry-1.docker.io
#14 DONE 0.0s

#12 exporting to image
ERROR: failed to receive status: rpc error: code = Unavailable desc = closing transport due to: connection error: desc = "error reading from server: EOF", received prior goaway: code: NO_ERROR, debug data: "graceful_stop"

Check build summary support
  Build summary requires a build reference
Error: buildx failed with: ERROR: failed to receive status: rpc error: code = Unavailable desc = closing transport due to: connection error: desc = "error reading from server: EOF", received prior goaway: code: NO_ERROR, debug data: "graceful_stop"
```
From the log, I noticed that the push produced an entry from a different container (***/app:pull)

After restarting the crashed container is built correctly.

This is the part of the yml that is used in the build:
```yml
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3

      - name: Login to Docker Hub
        uses: docker/login-action@v3
        with:
          username: ${{ secrets.username }}
          password: ${{ secrets.token }}
          logout: false

      - name: Build and push Docker image for ${{ inputs.serviceName }}
        uses: docker/build-push-action@v6
        with:
          push: true
          context: .
          file: ./${{ inputs.workingDirectory }}/Dockerfile
          platforms: linux/amd64, linux/arm64
          tags: ${{ secrets.username }}/${{ inputs.imageName }}:${{ vars.LABEL }}
```

Docker info:
```bash
docker info
Client: Docker Engine - Community
 Version:    27.3.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.17.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.7
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 2
 Server Version: 27.3.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7f7fdf5fed64eb6a7caf99b3e12efcf9d60e311c
 runc version: v1.1.14-0-g2c9f560
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 5.15.0-122-generic
 Operating System: Ubuntu 22.04.5 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 61.31GiB
 Name: server
 ID: 6c716f54-62b4-469e-9f99-aa2aabec961e
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

Can anyone explain what the error is (or how it appears) and how to fix it?

**Comment by crazy-max** [comment]
This looks similar to https://github.com/docker/build-push-action/issues/1257

Do you have a link to your repo or is it private?

**Comment by PanDeOps** [comment]
it's private repo

**Comment by erikburt** [comment]
We've seen a similar error in our public repo: https://github.com/smartcontractkit/chainlink/actions/runs/11808083477/job/32896022220

Only seen it for ARM64 builds (on Github-hosted ARM64 runners), I've also seen a few other errors.

**Comment by erikburt** [comment]
Here's an EOF error with Buildkit debug logs, although it appears different than the error that was originally posted about.

 https://github.com/smartcontractkit/chainlink/actions/runs/11863864029/job/33066447611?pr=15272#step:7:93

```
level=error msg="/moby.buildkit.v1.Control/Solve returned error: rpc error: code = Internal desc = rpc error: code = Internal desc = unexpected EOF" ...
Internal: rpc error: code = Internal desc = unexpected EOF
```

**Comment by crazy-max** [comment]
This is similar to https://github.com/docker/build-push-action/issues/1257, https://github.com/docker/build-push-action/issues/1263 and has been fixed in https://github.com/moby/buildkit/pull/5530.

You can either downgrade buildkit in your workflow:

```yaml
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3
        with:
          driver-opts: |
            image=moby/buildkit:v0.16.0
```

Or use a version from master with the fix in the meantime:

```yaml
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3
        with:
          driver-opts: |
            image=moby/buildkit:nightly@sha256:b26268202a6d69bed47856db1a625b3fde1c3bfb0bf38c0873496d6b5b0df6ca
```

**Comment by tri-bao** [comment]
We got this error on our own github runner. According to [buildkit/pull/5530](https://github.com/moby/buildkit/pull/5530), the fix should be available in buildx >= v0.17.2. We are using v0.19.2. @crazy-max is the fix part of v0.19.2?

I noticed that, github-manged runners use the same docker/buildx version as our self-hosted runners but we don't see the problem with github-managed runners, only happens to ours. The difference is that: in case of github-managed runners, each job runs in one isolated VM (one VM per job); in our self-hosted runners, we run several runner instances in the same VM so different build jobs share the same docker engine. Is that the problem? BTW, we use docker/bake-action@v5

Thanks for your advice

Below is the information extracted from the output of the failed workflow

```
Buildx version
  /usr/bin/docker buildx version
  github.com/docker/buildx v0.19.2 1fc5647dc281ca3c2ad5b451aeff2dce84f1dc49
```

```
Docker info
  /usr/bin/docker version
  Client: Docker Engine - Community
   Version:           26.1.3
   API version:       1.45
   Go version:        go1.21.10
   Git commit:        b72abbb
   Built:             Thu May 16 08:33:29 2024
   OS/Arch:           linux/amd64
   Context:           default
  
  Server: Docker Engine - Community
   Engine:
    Version:          26.1.3
    API version:      1.45 (minimum version 1.24)
    Go version:       go1.21.10
    Git commit:       8e96db1
    Built:            Thu May 16 08:33:29 2024
    OS/Arch:          linux/amd64
    Experimental:     false
   containerd:
    Version:          1.7.24
    GitCommit:        88bf19b2105c8b1[756](https://github.com/ecap-tech/eclipse/actions/runs/12594336969/job/35101703061#step:3:778)0993bee28a01ddc2f97182
   runc:
    Version:          1.2.2
    GitCommit:        v1.2.2-0-g7cb3632
   docker-init:
    Version:          0.19.0
    GitCommit:        de40ad0
  /usr/bin/docker info
  Client: Docker Engine - Community
   Version:    26.1.3
   Context:    default
   Debug Mode: false
   Plugins:
    buildx: Docker Buildx (Docker Inc.)
      Version:  v0.19.2
      Path:     /usr/libexec/docker/cli-plugins/docker-buildx
    compose: Docker Compose (Docker Inc.)
      Version:  v2.27.1
      Path:     /usr/libexec/docker/cli-plugins/docker-compose
  
  Server:
   Containers: 2
    Running: 1
    Paused: 0
    Stopped: 1
   Images: 16
   Server Version: 26.1.3
   Storage Driver: overlay2
    Backing Filesystem: extfs
    Supports d_type: true
    Using metacopy: false
    Native Overlay Diff: false
    userxattr: false
   Logging Driver: json-file
   Cgroup Driver: systemd
   Cgroup Version: 2
   Plugins:
    Volume: local
    Network: bridge host ipvlan macvlan null overlay
    Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
   Swarm: inactive
   Runtimes: io.containerd.runc.v2 runc
   Default Runtime: runc
   Init Binary: docker-init
   containerd version: 88bf19b2105c8b17560993bee28a01ddc2f97182
   runc version: v1.2.2-0-g7cb3632
   init version: de40ad0
   Security Options:
    apparmor
    seccomp
     Profile: builtin
    cgroupns
   Kernel Version: 6.5.0-1025-azure
   Operating System: Ubuntu 22.04.5 LTS
   OSType: linux
   Architecture: x86_64
   CPUs: 4
   Total Memory: 15.61GiB
   Name: shared-build-agents00001Q
   ID: b2edac28-[789](https://github.com/ecap-tech/eclipse/actions/runs/12594336969/job/35101703061#step:3:811)f-42dc-aded-0f242bd1033e
   Docker Root Dir: /var/lib/docker
   Debug Mode: false
   Experimental: false
   Insecure Registries:
    127.0.0.0/8
   Live Restore Enabled: false
```

**Comment by Yen** [comment]
We are seeing similar versions on recent versions of docker and buildx with self-hosted runners on Ubuntu 24.04. A rerun of the workflow fixes it in most cases, but obviously an issue for CI.
```
#6 [chef 1/2] FROM docker.io/lukemathwalker/cargo-chef:latest-rust-1@sha256:7ec09582d31061e406eb7542d4aa761e1e90f0296b80042ea3926e2176f3cbef
#6 extracting sha256:fbf93b646d6b4d6612a25d50599f4d8e3c7785f83c6ba085219f9d4d334e8aa3
ERROR: failed to receive status: rpc error: code = Unavailable desc = closing transport due to: connection error: desc = "error reading from server: EOF", received prior goaway: code: NO_ERROR, debug data: "graceful_stop"
Error: buildx failed with: ERROR: failed to receive status: rpc error: code = Unavailable desc = closing transport due to: connection error: desc = "error reading from server: EOF", received prior goaway: code: NO_ERROR, debug data: "graceful_stop"
```
```
Client: Docker Engine - Community
   Version:           27.5.1
   API version:       1.47
   Go version:        go1.22.11
   Git commit:        9f9e405
   Built:             Wed Jan 22 13:41:48 2025
   OS/Arch:           linux/amd64
   Context:           default
  
  Server: Docker Engine - Community
   Engine:
    Version:          27.5.1
    API version:      1.47 (minimum version 1.24)
    Go version:       go1.22.11
    Git commit:       4c9b3b0
    Built:            Wed Jan 22 13:41:48 2025
    OS/Arch:          linux/amd64
    Experimental:     false
   containerd:
    Version:          1.7.25
    GitCommit:        bcc810d6b9066471b0b6fa75f557a[15](https://github.com/Valinor-Research/valinor-rs/actions/runs/12964894558/job/36164083265#step:5:16)a1cbf31bb
   runc:
    Version:          1.2.4
    GitCommit:        v1.2.4-0-g6c52b3f
   docker-init:
    Version:          0.19.0
    GitCommit:        de40ad0
  /usr/bin/docker info
  Client: Docker Engine - Community
   Version:    27.5.1
   Context:    default
   Debug Mode: false
   Plugins:
    buildx: Docker Buildx (Docker Inc.)
      Version:  v0.20.0
      Path:     /usr/libexec/docker/cli-plugins/docker-buildx
    compose: Docker Compose (Docker Inc.)
      Version:  v2.32.4
      Path:     /usr/libexec/docker/cli-plugins/docker-compose
  
  Server:
   Containers: 1
    Running: 1
    Paused: 0
    Stopped: 0
   Images: 13
   Server Version: 27.5.1
   Storage Driver: overlay2
    Backing Filesystem: extfs
    Supports d_type: true
    Using metacopy: false
    Native Overlay Diff: true
    userxattr: false
   Logging Driver: json-file
   Cgroup Driver: systemd
   Cgroup Version: 2
   Plugins:
    Volume: local
    Network: bridge host ipvlan macvlan null overlay
    Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
   Swarm: inactive
   Runtimes: io.containerd.runc.v2 runc
   Default Runtime: runc
   Init Binary: docker-init
   containerd version: bcc810d6b9066471b0b6fa75f557a15a1cbf31bb
   runc version: v1.2.4-0-g6c52b3f
   init version: de40ad0
   Security Options:
    apparmor
    seccomp
     Profile: builtin
    cgroupns
   Kernel Version: 6.8.0-51-generic
   Operating System: Ubuntu 24.04.1 LTS
   OSType: linux
   Architecture: x86_64
   CPUs: [16](https://github.com/Valinor-Research/valinor-rs/actions/runs/12964894558/job/36164083265#step:5:17)
   Total Memory: 62.72GiB
   Name: gh-runner-1
   ID: 57363fa7-d476-47f2-9772-796448ecac6b
   Docker Root Dir: /var/lib/docker
   Debug Mode: false
   Experimental: false
   Insecure Registries:
    127.0.0.0/8
```
```
{
    "nodes": [
      {
        "name": "builder-0a48a65c-efd0-4cbb-9b65-84aa2e8a31b70",
        "endpoint": "unix:///var/run/docker.sock",
        "status": "running",
        "buildkitd-flags": "--allow-insecure-entitlement security.insecure --allow-insecure-entitlement network.host",
        "buildkit": "v0.18.2",
        "platforms": "linux/amd64,linux/amd64/v2,linux/amd64/v3,linux/386",
        "features": {
          "Automatically load images to the Docker Engine image store": true,
          "Cache export": true,
          "Docker exporter": true,
          "Multi-platform build": true,
          "OCI exporter": true
        },
        "labels": {
          "org.mobyproject.buildkit.worker.executor": "oci",
          "org.mobyproject.buildkit.worker.hostname": "1484ef91daf8",
          "org.mobyproject.buildkit.worker.network": "host",
          "org.mobyproject.buildkit.worker.oci.process-mode": "sandbox",
          "org.mobyproject.buildkit.worker.selinux.enabled": "false",
          "org.mobyproject.buildkit.worker.snapshotter": "overlayfs"
        },
        "gcPolicy": [
          {
            "all": false,
            "filter": [
              "type==source.local",
              "type==exec.cachemount",
              "type==source.git.checkout"
            ],
            "keepDuration": "48h0m0s"
          },
          {
            "all": false,
            "keepDuration": "1440h0m0s"
          },
          {
            "all": false
          },
          {
            "all": true
          }
        ]
      }
    ],
    "name": "builder-0a48a65c-efd0-4cbb-9b65-84aa2e8a31b7",
    "driver": "docker-container",
    "lastActivity": "[202](https://github.com/Valinor-Research/valinor-rs/actions/runs/12964894558/job/36164083265#step:5:208)5-01-25T12:26:10.000Z"
  }
```
Thanks
